File: src/commands/ping.py
from discord.ext import commands
from discord import app_commands, Interaction
from src.utils.classes import Custom_Command
from src.utils.consts import IDS




class MyCommand(Custom_Command):
  name = 'ping'


  @app_commands.command(
    name=name,
    description='Renvoi le délai de Quantum entre les messages et réponses'
  )
  async def fn(self, interaction: Interaction):

    await interaction.response.send_message(content='⌛ Loading...')
    websockets_latency = self.bot.ws.latency

    await interaction.edit_original_response(content=f"Discord Websocket ⇒ `{websockets_latency:.3f}ms`")




async def setup(bot):
  '''Cog setup'''
  await bot.add_cog(MyCommand(bot), guilds=[IDS.GUILDS.guild_main])

File: src/commands/ping.rs
use crate::utils::{Context, Error};



/// Display the latency of the bot
#[poise::command(
    slash_command,
    rename="ping",
)]
pub async fn cmd(ctx: Context<'_>) -> Result<(), Error> {

    let text = format!("⌛ Loading...");

    let start = std::time::Instant::now();
    let response = ctx.say(text).await?;
    let elapsed = start.elapsed();

    let text =  format!("Discord Websocket ⇒ `{}ms`", elapsed.as_millis());
    let msg = poise::CreateReply::default()
        .content(text);

    response.edit(ctx, msg).await?;

    Ok(())
}
